db/compaction_picker.cc

```diff
@@ -1128,6 +1134,25 @@ void LevelCompactionBuilder::SetupInitialFiles() {
     is_manual_ = true;
     parent_index_ = base_index_ = -1;
     PickFilesMarkedForCompaction();
+    if (start_level_inputs_.empty()) {
+      size_t i;
+      for (i = 0; i < vstorage_->BottommostFilesMarkedForCompaction().size();
+           ++i) {
+        auto& level_and_file =
+            vstorage_->BottommostFilesMarkedForCompaction()[i];
+        assert(!level_and_file.second->being_compacted);
+        start_level_inputs_.level = output_level_ = start_level_ =
+            level_and_file.first;
+        start_level_inputs_.files = {level_and_file.second};
+        if (compaction_picker_->ExpandInputsToCleanCut(cf_name_, vstorage_,
```

Comment:
Can you explain why you need this? If you are compacting only a bottom-most file, they don't have any other files to compact with, right? In all cases you'll only ever compact a single file, as I understand it?

Reply:
This diff is expected to compact Lmax-1 as well, if nothing is overlapping in the bottommost level.

Reply:
@igorcanadi - the files in bottom level can have the same user key at their endpoints. Example:
- file 1's last key is deletion of user key A with seqnum 20
- file 2's first key is put of user key A with seqnum 10
If they are compacted alone, compaction of file 1 will drop the deletion, and compaction of file 2 will keep the user key. We need to make sure file 1 and file 2 are picked together for compaction.

Reply:
Got it, good point :)